common/persistence/cassandra/cassandraQueue.go

```diff
@@ -364,16 +382,62 @@ func (q *cassandraQueue) UpdateAckLevel(
 }
 
 func (q *cassandraQueue) GetAckLevels() (map[string]int, error) {
-	queueMetadata, err := q.getQueueMetadata()
+	queueMetadata, err := q.getQueueMetadata(q.queueType)
 	if err != nil {
 		return nil, err
 	}
 
 	return queueMetadata.clusterAckLevels, nil
 }
 
-func (q *cassandraQueue) getQueueMetadata() (*queueMetadata, error) {
-	query := q.session.Query(templateGetQueueMetadataQuery, q.queueType)
+func (q *cassandraQueue) UpdateDLQAckLevel(
+	messageID int,
+	clusterName string,
+) error {
+
+	// Use negative queue type as the dlq type
+	queueMetadata, err := q.getQueueMetadata(-q.queueType)
+	if err != nil {
+		return &workflow.InternalServiceError{
+			Message: fmt.Sprintf("UpdateDLQAckLevel operation failed. Error %v", err),
+		}
+	}
+
+	// Ignore possibly delayed message
+	if queueMetadata.clusterAckLevels[clusterName] > messageID {
+		return nil
+	}
+
+	queueMetadata.clusterAckLevels[clusterName] = messageID
+	queueMetadata.version++
```

Comment:
If this should always be done together, consider making it a function and queueMetadata

Reply:
Moved to a common func